system/tortoise.go

```diff
@@ -8,9 +8,16 @@ import (
 
 //go:generate mockgen -package=mocks -destination=./mocks/tortoise.go -source=./tortoise.go
 
+// DecodedBallot contains all information required by tortoise.
+type DecodedBallot interface {
+	Opinion() types.Hash32
+	CancelWeight()
+	Store() error
+}
+
 // Tortoise is an interface provided by tortoise implementation.
 type Tortoise interface {
-	OnBallot(*types.Ballot)
+	DecodeBallot(*types.Ballot) (DecodedBallot, error)
```

Comment:
this interface (based on the usage in mesh.go) is not common in that DecodedBallot can be nil when error is nil. 
can you add comment to help caller when new code is written?